### PR TITLE
Add testing for mark command

### PR DESCRIPTION
**Pull request by jaylontan**

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        int bookingId = 1;
+        Status status = Status.COMPLETED;
+        String userInput = " " + PREFIX_BOOKING_ID + bookingId + " " + PREFIX_STATUS + status.name();
+
+        MarkCommand expectedCommand = new MarkCommand(bookingId, status);
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
```
Comment (pastchum) [inline src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java:25]: Good use of test cases to check the MarkCommandParser. Thorough and well-designed. LGTM!

File: src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        String userInput = " " + PREFIX_BOOKING_ID + "1" + " " + PREFIX_STATUS + "invalidStatus";
+        assertParseFailure(parser, userInput, MarkCommand.MESSAGE_INVALID_STATUS);
+    }
+
+    @Test
+    public void parse_extraPreamble_failure() {
+        String userInput = "randomText " + PREFIX_BOOKING_ID + "1" + " " + PREFIX_STATUS + "COMPLETED";
+        assertParseFailure(parser, userInput, String.format(
+                seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
+    }
```
Comment (pastchum) [inline src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java:58]: Good test cases, especially for extra edge cases such as extra preambles. LGTM!

**Review by pastchum — APPROVED** [review]

File: src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java (hunk 2 above)
Comment (LuBolin) [reply src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java:58]: I agree with pastchum! this is a lot of test cases, which helps to keep our program correct.

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    public void equals_sameObject_returnsTrue() {
+        MarkCommand command = new MarkCommand(1, Status.UPCOMING);
+        assertEquals(command, command);
+    }
+
+    @Test
+    public void equals_differentType_returnsFalse() {
+        MarkCommand command = new MarkCommand(1, Status.UPCOMING);
+        assertNotEquals(command, "Some String");
+    }
```
Comment (LuBolin) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:61]: Would this be necessary? Command does not inherit from string, so this can never happen I believe, and this test is not needed.

File: src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java (hunk 2 above)
Comment (isawangyx) [inline src/test/java/seedu/address/logic/parser/MarkCommandParserTest.java:58]: Nice work on covering the core functionality of MarkCommand! Perhaps you could consider adding a test case where the booking is already in the target status (e.g., marking an already COMPLETED booking as COMPLETED again), to verify that repeated status updates behave as expected.
